exporter/prometheus/prometheus.go

```diff
@@ -185,20 +185,27 @@ func (c *collector) Describe(ch chan<- *prometheus.Desc) {
 func (c *collector) Collect(ch chan<- prometheus.Metric) {
 	// We need a copy of all the view data up until this point.
 	viewData := c.cloneViewData()
-
 	for _, vd := range viewData {
 		sig := viewSignature(c.opts.Namespace, vd.View)
 		c.registeredViewsMu.Lock()
 		desc := c.registeredViews[sig]
 		c.registeredViewsMu.Unlock()
-
-		for _, row := range vd.Rows {
-			metric, err := c.toMetric(desc, vd.View, row)
+		if len(vd.Rows) == 0 {
+			metric, err := c.toMetric(desc, vd.View, &view.Row{Data: &view.CountData{}})
```

Comment:
How does this work when you switch to a different metric descriptor as soon as there is available data?

I don't know the canonical best approach here. OpenCensus is a dynamic instrumentation library which means you can introduce new aggregations in runtime. As far as I understand from your issue, Prometheus expects metric descriptors to be available even though there is no data collected on them. Maybe @fabxc can help us with https://github.com/census-instrumentation/opencensus-go/issues/935.

Reply:
The way I understand it the metric descriptor should not/does not change? Prometheus client libraries export default 0 values for metrics without labels. So I think the issue is that the metric is not sent during `Collect` if it doesn't have and data rows yet.

The Prometheus metrics should only be registered with the view once as far as I can tell.

Reply:
The point of the initialization recommendation is that all valid label combinations exist already. This will just expose a series with the same name but no labels combinations set at all. This is more harmful than not exposing any series since you know have a metric that changes its signature at runtime once data is observed.

See @negz second comment on the issue: https://github.com/census-instrumentation/opencensus-go/issues/935#issuecomment-424643888